build-ios.sh

```diff
@@ -1,69 +1,99 @@
 #!/bin/sh
 
+#  The expected iOS build invocation is:
+#    build-ios.sh [clean] [release|debug] ${ARCH} ${PLATFORM}
+#  where
+#    ARCH = arm64|arm64e|x86_64
+#    PLATFORM = iphoneos|iphonesimulator|xros|xrsimulator
+
 if [ "$1" == "clean" ]; then
- rm -f CMakeCache.txt *.cmake
- rm -rf out
- rm -rf .buildtools
-# make clean
+  echo "build-ios.sh: cleaning previous build artifacts"
+  rm -f CMakeCache.txt *.cmake
+  rm -rf out
+  rm -rf .buildtools
+#  make clean
+  shift
 fi
 
-if [ "$1" == "release" ] || [ "$2" == "release" ]; then
-BUILD_TYPE="Release"
-else
 BUILD_TYPE="Debug"
+if [ "$1" == "release" ]; then
+  BUILD_TYPE="Release"
+  shift
+elif [ "$1" == "debug" ]; then
+  BUILD_TYPE="Debug"
+  shift
 fi
 
 # Set Architecture: arm64, arm64e or x86_64
-if [ "$2" == "arm64" ] || [ "$3" == "arm64" ]; then
-IOS_ARCH="arm64"
-elif [ "$2" == "arm64e" ] || [ "$3" == "arm64e" ]; then
-IOS_ARCH="arm64e"
-else
-IOS_ARCH="x86_64"
+IOS_ARCH=$(/usr/bin/uname -m)
+if [ "$1" == "arm64" ]; then
+  IOS_ARCH="arm64"
+  shift
+elif [ "$1" == "arm64e" ]; then
+  IOS_ARCH="arm64e"
+  shift
+elif [ "$1" == "x86_64" ]; then
+  IOS_ARCH="x86_64"
+  shift
 fi
 
-# Set Platform: device or simulator
-if [ "$2" == "device" ] || [ "$3" == "device" ] || [ "$4" == "device" ]; then
-IOS_PLAT="iphoneos"
-else
+# the last param is expected to specify the platform name: iphoneos|iphonesimulator|xros|xrsimulator
+# so if it is non-empty and it is not "device", we take it as a valid platform name
+# otherwise we fall back to old iOS logic which only supported iphoneos|iphonesimulator
 IOS_PLAT="iphonesimulator"
+if [ -n "$1" ] && [ "$1" != "device" ]; then
```

Comment:
How can the first parameter denotes the architecture, build type and platform? You should have check for secound and third?

Reply:
'shift' command removes the consumed parameter, shifting $2 -> $1. so these params may be optional but are positional and the required order is: build-ios.sh [clean] [release|debug] ${ARCH} ${PLATFORM}